core/src/messages/state.rs

```diff
@@ -8,26 +8,36 @@ use serde::{Deserialize, Serialize};
 use std::net::IpAddr;
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct WorldStateMessage {
-    pub cluster: ClusterName,
-
-    #[serde(flatten)]
-    pub message: ClusterStateMessage,
+#[serde(untagged)]
+pub enum WorldStateMessage {
+    ClusterMessage {
+        cluster: ClusterName,
+        #[serde(flatten)]
+        message: ClusterStateMessage,
+    },
+    Heartbeat {
+        heartbeat: (),
```

Comment:
as per offline convo, make it so this serializes as {}

Reply:
I kept it as `null`, but made it future-compatible with a non-null payload.